### ISSUE TITLE
Suspicious move

### ISSUE THREAD

**Issue by Xazax-hun** [dotnet/coreclr#18872]
I think there might be a use after move in `src/jit/lower.cpp`

See: https://github.com/dotnet/coreclr/blob/master/src/jit/lower.cpp#L561

The variable `condRange` might have been moved just before its use.

**Comment by mikedn** [comment]
It's a bit confusing but that's not how it works. `std:move` does not move anything, it only enables movement. In this case the range is passed down to `InsertAfter` as an r-value reference and nothing ever get moved in this particular code.

And use after move is not incorrect in itself. Though in this case you'll end up with an assert in debug builds because `LastNode` becomes null and `fgSplitBlockAfterNode` does not like nulls.

The real question is why do r-value references need to be used here. AFAIR there was a reason for that but I can't remember it right now. It might have been done in an attempt to prevent shooting one's self foot rather than actual movement being needed.

**Comment by Xazax-hun** [comment]
Indeed, no actual move here, the rvalue reference used as a regular reference in the callee.
This is very confusing and prone to errors. Usually, rvalue reference means that we are free to "mess up" the objec's representation. This is not the case here, even though InsertAfter takes its argument as an rvalue reference, the referred object must stay intact. I think it would be great to have a comment to clarify this contract (and possibly document why do we need an rvalue reference in the first place).

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib
